File: python-api/src/api/handlers/polls/deletePoll/deletePollHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['POLLS_TABLE_NAME']
DYNAMO_POLLS_ID = os.environ['POLLS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def deletePoll(event, context):

    pk = event['pathParameters']['id']
    print('pk: ', pk)
    pk = DYNAMO_POLLS_ID + '#' + pk

    params = {
        'Key': {
            'PK': pk,
            'SK': pk
        }
    }

    try:
        table = dynamodb_client.Table(TABLE_NAME)
        response = table.delete_item(**params)
    except Exception as ex:
        return create_response(500, str(ex))

    print('response: ', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error deleting the Poll')

    return create_response(200, 'Poll with id: ' + pk + 'deleted successfully')

File: nodejs-api/src/api/handlers/polls/deletePoll/deletePollHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.POLLS_TABLE_NAME
const dynamoPollsId = process.env.POLLS_ID

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const deletePoll = async (event) => {
    const PK = event.pathParameters.id
    
    const params = {
      TableName: tableName,
      Key: {
        PK: dynamoPollsId + '#' + PK,
        SK: dynamoPollsId + '#' + PK
      }
    }
  
    try {
      const deletePoll = await db.delete(params).promise()
      console.log('deletePoll: ', deletePoll)
      return response(200, 'Poll with id: ' + PK + 'deleted successfully')
    } catch (err) {
      return response(500, 'Error deleting the Poll')
    }
}

module.exports = {
    deletePoll
}